.github/scripts/format_dbt_test_failures.py

```diff
@@ -524,24 +526,39 @@ def get_tablename_from_node(node: typing.Dict) -> str:
         return meta_tablename
 
     # Search for the model that is implicated in this test via the
-    # depends_on key
-    models = node.get("depends_on", {}).get("nodes", [])
-    if not models:
+    # test_metadata.kwargs.model attribute. Note that it is common to use the
+    # elements in the depends_on array for this purpose, but this approach
+    # is fraught, since the order of parents for tests with multiple
+    # dependencies is not clear and can differ:
+    # https://github.com/dbt-labs/dbt-core/issues/6746#issuecomment-1829860236
```

Comment:
Something about the way that `res_class_matches_pardat` performs its join breaks my previous assumption that the last element of `depends_on` always represents the model that the test is attached to (the workbook was reporting the source table for these tests as `pardat`). Using `test_metadata.kwargs.model` is a little bit more hacky, because it requires using a regex to extract the tablename from a string representing a `get_where_subquery()` call, but ultimately I think it's a more correct pointer to the source model than `depends_on`.